Model/IdealPaymentMethod.php

```diff
@@ -0,0 +1,212 @@
+            ],
+            'additionalInfo' => [
+                'type'  => 'select',
+                'field' => 'oxpayments__wdoxidee_additional_info',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_additional_info'),
+                'description' => Helper::translate('wd_config_additional_info_desc'),
+            ],
+            'deleteCanceledOrder' => [
+                'type' => 'select',
+                'field' => 'oxpayments__wdoxidee_delete_canceled_order',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_delete_cancel_order'),
+                'description' => Helper::translate('wd_config_delete_cancel_order_desc'),
+            ],
+            'deleteFailedOrder' => [
+                'type' => 'select',
+                'field' => 'oxpayments__wdoxidee_delete_failed_order',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_delete_failure_order'),
+                'description' => Helper::translate('wd_config_delete_failure_order_desc'),
+            ],
+        ];
+
+        return array_merge(parent::getConfigFields(), $aAdditionalFields);
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getCheckoutFields()
+    {
+        return [
+            'bank' => [
+                'type' => 'select',
+                'options' => $this->_aBankOptions,
+                'title' => Helper::translate('wd_ideal_legend'),
+            ],
+        ];
+    }
+
+    /**
+     * Returns array for bank select options
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getBanks()
+    {
+        //IdealBic extends Enum class and contains const variables which represent bank options.
+        //toArray() combines all const variables from IdealBic and converts them to array.
+        return IdealBic::toArray();
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getPublicFieldNames()
+    {
+        return array_merge(parent::getPublicFieldNames(), [
+            'descriptor',
+            'additionalInfo',
+            'deleteCanceledOrder',
+            'deleteFailedOrder',
+        ]);
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @param string $sAction
+     *
+     * @return SepaCreditTransferPaymentMethod
+     *
+     * @since 1.2.0
+     */
+    public function getPostProcessingPaymentMethod($sAction = '')
```

Comment:
I don't think we need the default value at this point. `wdpd-41-eps-payment-method` also does not have it.

Reply:
:+1: